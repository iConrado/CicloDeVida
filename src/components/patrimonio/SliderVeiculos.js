import React from 'react';

import { 
  View, 
  Text,
  Image, } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';
import monetizar from '../functions/monetizar';

const imgCar = require('../../imgs/ic_car_white.png');

export default class SliderVeiculos extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      tmpVeiculos: props.inicial 
    }; 
  }

  render() {
    return (
      <View style={[styles.viewVertical, styles.espacador]}>
        <Text style={styles.label}>Veículos (valor total - financiados ou não):</Text>
        <View style={styles.viewHorizontal}>
          <View style={styles.viewIcone}>
            <Image 
              style={styles.imgIcone}
              source={imgCar}
            />
          </View>
          <View style={styles.viewPosIcone}>
            <Slider
              style={styles.patrim_slIcone}
              minimumValue={0}
              maximumValue={500000}
              step={2000}
              minimumTrackTintColor='#14567A'
              thumbTintColor='#14567A'
              value={this.state.tmpVeiculos}
              onValueChange={(value) => this.setState({ tmpVeiculos: value })}
              onSlidingComplete={(value) => this.props.retorno(value)}
            />
            <Text style={styles.patrim_txIcone}>{monetizar(this.state.tmpVeiculos)}</Text>
          </View>
        </View>
      </View>
    );
  }
}
