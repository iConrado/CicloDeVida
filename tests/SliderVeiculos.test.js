import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import SliderVeiculos from '../src/components/patrimonio/SliderVeiculos';


it('renders correctly', () => {
  const rendered = renderer.create(<SliderVeiculos />).toJSON();
  expect(rendered).toMatchSnapshot();
});
