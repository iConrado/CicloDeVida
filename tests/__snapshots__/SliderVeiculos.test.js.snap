// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderVeiculos renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderVeiculos
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          style={
            Object {
              "color": "#666",
            }
          }
        >
          Veículos (valor total - financiados ou não):
        </Text>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={500000}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={2000}
              style={
                Object {
                  "height": 40,
                  "width": "65%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <Text
              style={
                Object {
                  "height": 40,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "35%",
                }
              }
            >
              R$ 50
            </Text>
          </View>
        </View>,
      ],
      "style": Array [
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
        Object {
          "marginTop": 15,
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Veículos (valor total - financiados ou não):",
          "style": Object {
            "color": "#666",
          },
        },
        "ref": null,
        "rendered": "Veículos (valor total - financiados ou não):",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
            </View>,
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                }
              }
            >
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={500000}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={2000}
                style={
                  Object {
                    "height": 40,
                    "width": "65%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />
              <Text
                style={
                  Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  }
                }
              >
                R$ 50
              </Text>
            </View>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />,
              "style": Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {
                  "height": 35,
                  "width": 35,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Slider
                  animationType="timing"
                  debugTouchArea={false}
                  maximumTrackTintColor="#b3b3b3"
                  maximumValue={500000}
                  minimumTrackTintColor="#14567A"
                  minimumValue={0}
                  onSlidingComplete={[Function]}
                  onValueChange={[Function]}
                  step={2000}
                  style={
                    Object {
                      "height": 40,
                      "width": "65%",
                    }
                  }
                  thumbTintColor="#14567A"
                  thumbTouchSize={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                  value={50}
                />,
                <Text
                  style={
                    Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    }
                  }
                >
                  R$ 50
                </Text>,
              ],
              "style": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animationType": "timing",
                  "debugTouchArea": false,
                  "maximumTrackTintColor": "#b3b3b3",
                  "maximumValue": 500000,
                  "minimumTrackTintColor": "#14567A",
                  "minimumValue": 0,
                  "onSlidingComplete": [Function],
                  "onValueChange": [Function],
                  "step": 2000,
                  "style": Object {
                    "height": 40,
                    "width": "65%",
                  },
                  "thumbTintColor": "#14567A",
                  "thumbTouchSize": Object {
                    "height": 40,
                    "width": 40,
                  },
                  "value": 50,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "R$ 50",
                  "style": Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  },
                },
                "ref": null,
                "rendered": "R$ 50",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Veículos (valor total - financiados ou não):
          </Text>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                }
              }
            >
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={500000}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={2000}
                style={
                  Object {
                    "height": 40,
                    "width": "65%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />
              <Text
                style={
                  Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  }
                }
              >
                R$ 50
              </Text>
            </View>
          </View>,
        ],
        "style": Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          },
          Object {
            "marginTop": 15,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Veículos (valor total - financiados ou não):",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Veículos (valor total - financiados ou não):",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#48AEB5",
                    "borderRadius": 13,
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 35,
                      "width": 35,
                    }
                  }
                />
              </View>,
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "paddingLeft": 10,
                  }
                }
              >
                <Slider
                  animationType="timing"
                  debugTouchArea={false}
                  maximumTrackTintColor="#b3b3b3"
                  maximumValue={500000}
                  minimumTrackTintColor="#14567A"
                  minimumValue={0}
                  onSlidingComplete={[Function]}
                  onValueChange={[Function]}
                  step={2000}
                  style={
                    Object {
                      "height": 40,
                      "width": "65%",
                    }
                  }
                  thumbTintColor="#14567A"
                  thumbTouchSize={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                  value={50}
                />
                <Text
                  style={
                    Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    }
                  }
                >
                  R$ 50
                </Text>
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  source={1}
                  style={
                    Object {
                      "height": 35,
                      "width": 35,
                    }
                  }
                />,
                "style": Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {
                    "height": 35,
                    "width": 35,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Slider
                    animationType="timing"
                    debugTouchArea={false}
                    maximumTrackTintColor="#b3b3b3"
                    maximumValue={500000}
                    minimumTrackTintColor="#14567A"
                    minimumValue={0}
                    onSlidingComplete={[Function]}
                    onValueChange={[Function]}
                    step={2000}
                    style={
                      Object {
                        "height": 40,
                        "width": "65%",
                      }
                    }
                    thumbTintColor="#14567A"
                    thumbTouchSize={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                    value={50}
                  />,
                  <Text
                    style={
                      Object {
                        "height": 40,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                        "width": "35%",
                      }
                    }
                  >
                    R$ 50
                  </Text>,
                ],
                "style": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "animationType": "timing",
                    "debugTouchArea": false,
                    "maximumTrackTintColor": "#b3b3b3",
                    "maximumValue": 500000,
                    "minimumTrackTintColor": "#14567A",
                    "minimumValue": 0,
                    "onSlidingComplete": [Function],
                    "onValueChange": [Function],
                    "step": 2000,
                    "style": Object {
                      "height": 40,
                      "width": "65%",
                    },
                    "thumbTintColor": "#14567A",
                    "thumbTouchSize": Object {
                      "height": 40,
                      "width": 40,
                    },
                    "value": 50,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "R$ 50",
                    "style": Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    },
                  },
                  "ref": null,
                  "rendered": "R$ 50",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
